### PR TITLE
Add updates for numeric IntPtr

### LINKED ISSUE #29625: [17.3] numeric IntPtr
`System.IntPtr` and `System.UIntPtr` become numeric types, aliased by `nint` and `nuint`, and gain the built-in status of `nint` and `nuint`.

- *Championed issue* dotnet/csharplang#6065
- [*Feature spec*](https://github.com/dotnet/csharplang/blob/main/proposals/numeric-intptr.md)

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #29625

The types `nint` and `nuint` are now aliases for `System.IntPtr` and `System.UIntPtr`, respectively.

These updates mostly involve removing text that described how the `nint` and `nuint` types behaved differently than the underlying types. They now behave the same.

### CODE REVIEW

File: docs/csharp/language-reference/builtin-types/integral-numeric-types.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ f1_keywords:
   - "uint_CSharpKeyword"
   - "long_CSharpKeyword"
   - "ulong_CSharpKeyword"
+  - nint_CSharpKeyword
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/builtin-types/integral-numeric-types.md:14]: Thanks! good catch.

[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/builtin-types/integral-numeric-types.md]: > Also, "nint" is only a contextual keyword. Does that matter for this label?

Nope. These define the tokens for routing F1 commands from Visual Studio. VS passes along that token, and the docs site routes to the page that has the token defined.

File: docs/csharp/whats-new/csharp-11.md [hunk 2]
```diff
@@ -110,6 +109,10 @@ Generic math created other requirements on the language.
 - *relaxed shift operator requirements*: C# 11 removes the requirement that the second operand must be an `int` or implicitly convertible to `int`. This allows types that implement generic math interfaces to be used in these locations.
 - *checked and unchecked user defined operators*: Developers can now define `checked` and `unchecked` arithmetic operators. The compiler generates calls to the correct variant based on the current context. You can read more about `checked` operators in the article on [Arithmetic operators](../language-reference/operators/arithmetic-operators.md).
 
+## Numeric `IntPtr` and `UintPtr`
```
Comment (jcouv) [inline docs/csharp/whats-new/csharp-11.md:112]: >UintPtr [](http://example.com/codeflow?start=25&length=7)

capitalization seems wrong: should be "UIntPtr"

**Comment by jcouv** [comment]
nit: OP has a typo. NuintPtr->UIntPtr
> The types nint and nuint are now aliases for System.IntPtr and System.NuintPtr, respectively.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3) modulo remaining typo for UIntPtr

**Review by adegeo — APPROVED** [review]
LGTM! Very clear.

**Pull request merged**